build.gradle

```diff
@@ -256,3 +259,70 @@ afterEvaluate {
         reportUndocumented = false
     }
 }
+
+allprojects {
+    apply plugin: "jacoco"
+
+    jacoco {
+        toolVersion = "0.8.5"
+        reportsDir = file("${buildDir}/jacoco-reports")
+    }
+}
+
+subprojects {
+    task testCoverage(type: JacocoReport, dependsOn: jvmTest) {
+        group = "Reporting"
+        description = "Generate Jacoco coverage reports."
+
+        def coverageSourceDirs = [
+            "common/src",
+            "jvm/src"
+        ]
+        classDirectories.from files(fileTree(dir: "${buildDir}/classes/kotlin/jvm/"))
+        sourceDirectories.from files(coverageSourceDirs)
+        additionalSourceDirs.from files(coverageSourceDirs)
+        executionData.from files("${buildDir}/jacoco/jvmTest.exec")
+
+        reports {
+            xml.enabled true
+            csv.enabled false
+            html.enabled true
+
+            html.destination file("${buildDir}/jacoco-reports/html")
```

Comment:
Doesn't Gradle have reports directory for such purposes?

Reply:
I faced an issue when Gradle overwrite this directory in parrallel